pythonforandroid/toolchain.py

```diff
@@ -227,6 +226,249 @@ def split_argument_list(arg_list):
     return re.split(r'[ ,]+', arg_list)
 
 
+def __expand_requirements_arg_from_project_files(ctx, args):
+    """Parse additional requirements from setup.py or pyproject.toml file
+    and add to --requirements arg if --use_setup_py argument was specified"""
+    all_recipes = [
+        recipe.lower() for recipe in
+        set(Recipe.list_recipes(ctx))
+    ]
+
+    has_setup_py_or_toml = False
+    if getattr(args, "private", None) is not None:
+        project_dir = getattr(args, "private")
+        has_setup_py = os.path.exists(
+            os.path.join(project_dir, "setup.py"))
+        has_toml = os.path.exists(
+            os.path.join(project_dir, "pyproject.toml"))
+        has_setup_py_or_toml = has_setup_py or has_toml
+
+    # Add dependencies from setup.py, but only if they are recipes
+    # (because otherwise, setup.py itself will install them later)
+    if has_setup_py_or_toml and getattr(args, "use_setup_py", False):
+        try:
+            info("Analyzing package dependencies. MAY TAKE A WHILE.")
+            # Get all the dependencies corresponding to a recipe:
+            dependencies = [
+                dep.lower() for dep in
+                get_dep_names_of_package(
+                    args.private,
+                    keep_version_pins=True,
+                    recursive=True,
+                    verbose=True,
+                )
+            ]
+            info("Dependencies obtained: " + str(dependencies))
+            dependencies = set(dependencies).intersection(
+                set(all_recipes)
+            )
+
+            # Add dependencies to argument list:
+            if len(dependencies) > 0:
+                if len(args.requirements) > 0:
+                    args.requirements += u","
+                args.requirements += u",".join(dependencies)
+
+        except ValueError:
+            # Not a python package, apparently.
+            warning(
+                "Processing failed, is this project a valid "
+                "package? Will continue WITHOUT setup.py deps."
+            )
+
+
+def has_a_recipe(ctx, requirement):
+    all_recipes = [
+        recipe.lower() for recipe in
+        set(Recipe.list_recipes(ctx))
+    ]
+    requirement_name = re.sub(r'==\d+(\.\d+)*', '', requirement)
```

Comment:
Here, into this regex, we shouldn't only account for digits.
`requirement_name = re.sub(r'==.+', '', requirement)` or `requirement.split('==')[0]`should be good.

As an example: `kivy==master` is a pretty common use case and should be covered.

Reply:
Ah ok.  I was not aware of this.  I use this regex pattern in 3 places apparently.  Would replacing them with `==[^\s]+`, to include anything but whitespaces, do the trick?

Reply:
Uhm, why whitespaces shouldn't be considered? I'm missing some context?